src/Motors/MotorModule.c

```diff
@@ -0,0 +1,262 @@
+	}
+}
+
+void indicatorRight(){
+	currentIndicater = millis();
+	if((currentIndicater - previousIndicater) >= blinkInterval){
+		previousIndicater = currentIndicater;
+		PORTC ^= (1<<DDC2); //PC2
+	}
+}
+
+void resetIndicator(){
+	PORTC &= ~(1<<PORTC1);
+	PORTC &= ~(1<<PORTC2);
+}
+
+/* ========================= Speed/Direction functions ================================= */
+
+void stopMotors(){
+	OCR1A = 0;	// PB1
+	OCR1B = 0;  // PB2
+}
+
+// MotorSpeeds should always be between 0 - 255 | The range of the MotorSpeed
+// Sets the speed on both motors at once
+void setAllMotorSpeed(uint8_t newSpeed){
+	resetIndicator();
+	OCR1A = newSpeed;
+	OCR1B = newSpeed;
+}
+
+// Only changes the speed in the left motor side
+// Without influencing the other motor
+void setLeftMotorSpeed(uint8_t newSpeed){
+	OCR1A = newSpeed;
+}
+
+// Only changes the speed in the right motor side
+// Without influencing the other motor
+void setRightMotorSpeed(uint8_t newSpeed){
+	OCR1B = newSpeed;
+}
+
+/* ========================= Direction functions ================================= */
+
+// Sets the driving direction of the car
+void setMotorDirection(bool direction){
+	// forward direction
+	// true is forward
+	if(direction){
+		PORTB &= ~(1<<DDB0);
+		PORTB &= ~(1<<DDB3);
+	}
+	// backward direction
+	// false is backwards
+	else if(!direction){
+		PORTB |= (1<<DDB0);
+		PORTB |= (1<<DDB3);
+	}
+}
+
+void setIndividualDirection(bool leftDirection, bool rightDirection){
+	if(leftDirection){
+		PORTB &= ~(1<<DDB0);
+	} else {
+		PORTB &= (1<<DDB0);
+	}
+	
+	if(rightDirection){
+		PORTB &= ~(1<<DDB3);
+	} else {
+		PORTB |= (1<<DDB3);
+	}
+}
+
+/* ========================= Steering functions ================================= */
+
+// Sets the leftMotorForce to some speed and sets the right motor to 0
+void steerLeftSimple(uint8_t leftMotorForce){
+	setLeftMotorSpeed(leftMotorForce);
+	setRightMotorSpeed(0);
+	indicatorLeft();
+}
+
+// Sets the rightMotorForce to some speed and sets the right motor to 0
+void steerRightSimple(uint8_t rightMotorForce){
+	setRightMotorSpeed(rightMotorForce);
+	setLeftMotorSpeed(0);
+	indicatorRight();
+}
+
+void zeroDegreeLeft(uint8_t motorSpeed){
+	setIndividualDirection(false, true);
+	setAllMotorSpeed(motorSpeed);
+	indicatorLeft();
+}
+
+void zeroDegreeRight(uint8_t motorSpeed){
+	setIndividualDirection(true, false);
+	setAllMotorSpeed(motorSpeed);
+	indicatorRight();
+}
+
+// Complex steering takes the individual speeds for both sides
+// because of that more complex steering maneuvers are possible
+void complexSteering(uint8_t rightMotorForce, uint8_t leftMotorForce){
+	setLeftMotorSpeed(leftMotorForce);
+	setRightMotorSpeed(rightMotorForce);
+}
+
+/* ========================= Testing functions ================================= */
+
+void motorLinearIncreaseTest(){
+	// While driving forward increases the speed linear over time
+	setMotorDirection(true);
+	for(int i = 0; i <= 255;){
+		setAllMotorSpeed(i);
+		_delay_ms(100);
```

Comment:
I personally always get errors when coding with the _delay_ms() function shouldn't we use interrupts for this?

Reply:
Inside the init function I added a note regarding this. The delay function didn't produce any errors but the timing wasn't correct. Therefore the functions need to be revisited to change it. Wasn't sure to do it now or save the time to start on the sonar's